### DISCUSSION TITLE
Cardmod help

### DISCUSSION THREAD

**Discussion by ojm88** [punxaphil/custom-sonos-card#538] [Q&A]
hi guys

Can someone help me with how I can beat the same / integrate both cardmods in the same "card" I attach a picture of the codes.![image](https://github.com/user-attachments/assets/425f1a1d-2d37-4f23-a651-45988ea20adc)

**Comment by punxaphil** [comment]
Paste the yaml as text here please

**Reply by ojm88** [reply]
card_mod:
  style:
    sonos-player$: |
      .container {        
        grid-template-areas:
            'header artwork'
            'controls artwork' !important;
        grid-template-columns: 2fr 1fr !important;
      }
    .: ""

**Reply by ojm88** [reply]
card_mod:
      style:
        sonos-player:
          $:
            sonos-player-header$: |
              .info .song {
                color: var(--primary-text-color);
                font-weight: 600;
                font-size: 25px;
              }
        .: |
          ha-card {
            --primary-color: none;
            --secondary-background-color: none;
          }

**Reply by ojm88** [reply]
> Paste the yaml as text here please

Hello.
Did you figure out how to combine the cardmod codes?

**Reply by punxaphil** [reply]
```
type: custom:sonos-card
card_mod:
  style:
    sonos-player$: |
      .header {
        max-width: 200px
      }
      .container {        
        grid-template-areas:
            'header artwork'
            'controls artwork' !important;
        grid-template-columns: 2fr 1fr !important;
      }

    sonos-player$ sonos-player-header$: |
      .info .song {
        color: var(--primary-text-color);
        font-weight: 600;
        font-size: 25px;
      }
    .: |
      ha-card {
        --primary-color: none;
        --secondary-background-color: none;
      }
```

**Reply by ojm88** [reply]
> ```
> 
> type: custom:sonos-card
> 
> card_mod:
> 
>   style:
> 
>     sonos-player$: |
> 
>       .header {
> 
>         max-width: 200px
> 
>       }
> 
>       .container {        
> 
>         grid-template-areas:
> 
>             'header artwork'
> 
>             'controls artwork' !important;
> 
>         grid-template-columns: 2fr 1fr !important;
> 
>       }
> 
> 
> 
>     sonos-player$ sonos-player-header$: |
> 
>       .info .song {
> 
>         color: var(--primary-text-color);
> 
>         font-weight: 600;
> 
>         font-size: 25px;
> 
>       }
> 
>     .: |
> 
>       ha-card {
> 
>         --primary-color: none;
> 
>         --secondary-background-color: none;
> 
>       }
> 
> ```

Fantastic. Then I've learned something new.

Can you help me further with getting the information (song title etc.) to use the entire space at the top of the card?

I would also like the album cover to be moved down.

Thanks for all your help.![image](https://github.com/user-attachments/assets/054b5928-0889-4b50-9796-49ab44eba2dc)

**Reply by punxaphil** [reply]
```
type: custom:sonos-card
card_mod:
  style:
    sonos-player$: |
      .container {        
        grid-template-areas:
            'header header'
            'controls artwork'
            'controls _'
            'controls _'
            'controls _' !important;
        grid-template-columns: 2fr 1fr !important;
      }
```

**Reply by ojm88** [reply]
> ```
> 
> type: custom:sonos-card
> 
> card_mod:
> 
>   style:
> 
>     sonos-player$: |
> 
>       .container {        
> 
>         grid-template-areas:
> 
>             'header header'
> 
>             'controls artwork'
> 
>             'controls _'
> 
>             'controls _'
> 
>             'controls _' !important;
> 
>         grid-template-columns: 2fr 1fr !important;
> 
>       }
> 
> ```

Nice.
What am I doing wrong since I can't get the album cover bigger?
This is the last thing left :-)

![image](https://github.com/user-attachments/assets/980b2172-4ffd-45da-a76e-037532ce47c8)

**Reply by punxaphil** [reply]
```
type: custom:sonos-card
card_mod:
  style:
    sonos-player$: |
      .container {        
        grid-template-areas:
            'header header'
            'controls artwork'
            'controls _'
            'controls _'
            'controls _' !important;
        grid-template-columns: 1fr 1fr !important;
      }
      .artwork {
        min-height: 8rem !important;
      }
    sonos-player$ sonos-player-header$: |
      .info .song {
        color: var(--primary-text-color);
        font-weight: 600;
        font-size: 25px;
      }
    .: |
      ha-card {
        --primary-color: none;
        --secondary-background-color: none;
      }
```

**Reply by ojm88** [reply]
> ```
> 
> type: custom:sonos-card
> 
> card_mod:
> 
>   style:
> 
>     sonos-player$: |
> 
>       .container {        
> 
>         grid-template-areas:
> 
>             'header header'
> 
>             'controls artwork'
> 
>             'controls _'
> 
>             'controls _'
> 
>             'controls _' !important;
> 
>         grid-template-columns: 1fr 1fr !important;
> 
>       }
> 
>       .artwork {
> 
>         min-height: 8rem !important;
> 
>       }
> 
>     sonos-player$ sonos-player-header$: |
> 
>       .info .song {
> 
>         color: var(--primary-text-color);
> 
>         font-weight: 600;
> 
>         font-size: 25px;
> 
>       }
> 
>     .: |
> 
>       ha-card {
> 
>         --primary-color: none;
> 
>         --secondary-background-color: none;
> 
>       }
> 
> ```

fantastic.

Now I think the Sonos card will be perfect for IOS devices :)
![image](https://github.com/user-attachments/assets/b89b021a-fb59-4569-b2cb-459b55aba61e)

**Reply by punxaphil** [reply]
👍


---
PunxaPhil
[![image](upload://uFDZrifftWGk7LL8U9kgEmWyt1.png)](https://patreon.com/PunxaPhil) [![image](upload://d8Ke1WwUUcaZkLgHjhVbs5trVjM.png)](https://www.buymeacoffee.com/punxaphil)
